audit/audit.go

```diff
@@ -0,0 +1,272 @@
+/*
+Package audit implements a smart client to a remote audit server. This handles any connection problems silently,
+while providing ways to detect problems through whatever alerting method is desired. It also handles an exponential
+backoff when trying to reconnect to the remote audit server as to prevent any overwhelming of the remote audit server.
+
+This is the preferred way to connect to a remote audit server. There is a more low-level way to connect to a remote
+audit server in the base package. This one allows for you to completely customize how you want to handle issues,
+but it is more work to do so.
+
+Example using the domainsocket package:
+
+	// Create a function that will create a new connection to the remote audit server.
+	// We use this function to create a new connection when the connection is broken.
+	cc := func() (conn.Audit, error) {
+		return conn.NewDomainSocket()
+	}
+
+	c, err := audit.New(cc)
+	if err != nil {
+		// Handle error.
+	}
+	defer c.Close(context.Background())
+
+	// This is optional if you want to get notifications of logging problems.
+	go func() {
+		for notifyMsg := range c.Notify() {
+			// Handle error notification.
+			// You can log them or whatever you want to do.
+		}
+	}()
+
+	// Send a message to the remote audit server.
+	if err := c.Send(context.Background(), msgs.Msg{<add record information>}); err != nil {
+		// Handle error.
+		// Errors here will either be:
+		// 1. base.ErrValidation , which means your message is invalid.
+		// 2. A standard error, which means there is an error condition we haven't categorized yet.
```

Comment:
This second case is (in theory) unexpected right? Worth noting: "If this happens, please raise an issue"?